src/main/java/org/apache/commons/lang3/StringUtils.java

```diff
@@ -7434,7 +7434,7 @@ public static String rotate(final String str, final int shift) {
             return ArrayUtils.EMPTY_STRING_ARRAY;
         }
         final char[] c = str.toCharArray();
-        final List<String> list = new ArrayList<>();
+        final SplitBuffer buffer = SPLIT_BUFFER_THREAD_LOCAL.get().getBuffer();
```

Comment:
Leaving the variable name `list` would have reduced the changes to review I think?

Reply:
The SplitBuffer instance is not a List instance anymore, leave it with a name "list" would be misleading I think.